app/src/main/java/com/example/erabook/firebaseActivities/LogIn.kt

```diff
@@ -0,0 +1,75 @@
+package com.example.erabook.firebaseActivities
+
+import android.content.Intent
+import android.os.Bundle
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.navigation.findNavController
+import androidx.navigation.fragment.NavHostFragment
+import com.example.erabook.MainActivity
+import com.example.erabook.R
+import com.example.erabook.databinding.ActivityLogInBinding
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.ktx.Firebase
+
+class LogIn : AppCompatActivity() {
+    private lateinit var binding: ActivityLogInBinding
+    private lateinit var auth: FirebaseAuth
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityLogInBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        auth = Firebase.auth
+        binding.apply {
+            loginButton.setOnClickListener {
```

Comment:
Please create a method called `setupOnClickListeners()` and then inside that method perform all the click listeners.